build/get_buildvers.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import xml.etree.ElementTree as ET
+
+def _get_expression(pomFile, expression, logger = None):
+    pom = ET.parse(pomFile)
+    ns = {"pom": "http://maven.apache.org/POM/4.0.0"}
+    releases = []
+    release_prefix = "release"
+    for profile in pom.findall(".//pom:profile/pom:id", ns):
+       if profile.text.startswith(release_prefix):
+            releases.append(profile.text[len(release_prefix):])
+    snapshots = []
+    no_snapshots = []
+
+    for release in releases:
+        spark_version = pom.find(".//pom:spark{}.version".format(release), ns)
+        if (spark_version.text.endswith("SNAPSHOT")):
+            snapshots.append(release)
+        else:
+            no_snapshots.append(release)
+    excluded_shims = pom.find(".//pom:dyn.shim.excluded.releases", ns).text
+    if (excluded_shims):
+        for removed_shim in [x.strip() for x in excluded_shims.split(",")]:
+            if (removed_shim not in snapshots and removed_shim not in no_snapshots):
```

Comment:
nit: this looks insufficient as snapshots and no_snapshots should not have common items, can we try
```
if removed_shim in snapshots:
    snapshots.remove(removed_shim)
elif removed_shim in no_snapshots:
    no_snapshots.remove(removed_shim)
else:
    raise Exception('...')
```

Reply:
I was trying to guard against a fictitious shim that doesn't exist in either snap or no_snaps but is listed in the excluded_shims.